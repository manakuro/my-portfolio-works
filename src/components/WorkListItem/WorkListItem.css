.work-list-item-Card {
  border: 1px solid rgba(0, 0, 0, 0.1); }
  .work-list-item-Card_Img {
    cursor: pointer; }
  .work-list-item-Card_Desc {
    display: flex;
    flex-direction: column;
    height: 160px;
    padding: 12px; }
  .work-list-item-Card_DescHeading {
    margin-bottom: 10px;
    color: rgba(47,47,47,.9);
    line-height: 1.5;
    font-size: 1rem; }
  .work-list-item-Card_DescSubHeading {
    color: rgba(0, 0, 0, 0.54);
    font-size: .9rem;
    line-height: 1.5; }

.work-list-item .work-list-item-TechList {
  margin-top: auto;
  font-size: 1.2rem;
  display: flex; }
  .work-list-item .work-list-item-TechList .__react_component_tooltip.type-success {
    background-color: #06dca3;
    font-weight: bold;
    color: #fff; }

.work-list-item .language-icon {
  margin-right: 10px; }
