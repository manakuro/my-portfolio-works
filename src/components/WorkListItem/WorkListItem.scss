.work-list-item {

  &-Card {
    border: 1px solid rgba(0,0,0,.1);

    &_Img {
      cursor: pointer;
    }

    &_Desc {
      display: flex;
      flex-direction: column;
      height: 160px;
      padding: 12px;
    }

    &_DescHeading {
      margin-bottom: 10px;
      color: #2F2F2F;
      line-height: 1.5;
      font-size: 1rem;
    }

    &_DescSubHeading {
      color: rgba(0, 0, 0, 0.54);
      font-size: .9rem;
      line-height: 1.5;
    }
  }

  .work-list-item-TechList {
    margin-top: auto;
    font-size: 1.2rem;
    display: flex;

    // overwrite style of React Tooltip
    .__react_component_tooltip.type-success {
      background-color: #06dca3;
      font-weight: bold;
      color: #fff;
    }
  }

  .language-icon {
    margin-right: 10px;
  }
}
