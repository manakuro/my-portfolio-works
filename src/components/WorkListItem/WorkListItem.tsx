import * as React from 'react'

import { Work, Language } from '@/modules/home/reducer'

export interface WorkListItemProps {
  work: Work
  updateTargetWork: (payload: Work) => any
  updateCircle: HomeDispatchFromProps['updateCircle']
  languages: Language[]
  history?: History
}

import './WorkListItem.css'
import LanguageIcon from '@/components/LanguageIcon'
import { History } from 'history'
import { HomeDispatchFromProps } from '@/components/Home/Home'

export default class WorkListItem extends React.PureComponent<
  WorkListItemProps
> {
  constructor(props: WorkListItemProps) {
    super(props)
  }

  public render(): JSX.Element {
    const { work } = this.props

    const languageIcons = work.languages.map(l => {
      const language = this.getLanguage(l)
      return language ? <LanguageIcon language={language} key={l} /> : null
    })

    return (
      <div className="work-list-item">
        <div className="work-list-item-Card">
          <img
            src={work.img}
            onClick={this.handleClick}
            className="work-list-item-Card_Img"
          />

          <div className="work-list-item-Card_Desc">
            <h3 className="work-list-item-Card_DescHeading">{work.title}</h3>
            <p className="work-list-item-Card_DescSubHeading">
              {work.description}
            </p>

            <div className="work-list-item-TechList">{languageIcons}</div>
          </div>
        </div>
      </div>
    )
  }

  private getLanguage(id: number): Language | undefined {
    return this.props.languages.find(l => l.id === id)
  }

  private handleClick = (e: React.SyntheticEvent<EventTarget>): void => {
    const nativeEvent: any = e.nativeEvent
    const { pageX, pageY } = nativeEvent

    const marginTop = 130
    const circleStyle = {
      top: `${pageY - (marginTop + 50)}px`,
      left: `${pageX - 50}px`,
    }

    this.props.updateCircle(circleStyle)

    if (this.props.history)
      this.props.history.push({
        pathname: `/works/${this.props.work.id}`,
      })
  }
}
