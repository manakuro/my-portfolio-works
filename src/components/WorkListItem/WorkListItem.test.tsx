import * as React from 'react'
import { shallow } from 'enzyme'
import * as WorkListItem from './WorkListItem'
import { WorkListItemPropsWithCompose } from '@/components/WorkListItem/WorkListItem'

const props: WorkListItemPropsWithCompose = {
  work: {
    id: 1,
    title: 'EC Website',
    description: `
      Iâ€™ve been a CMT for ten years now, and I have, literally and
      figuratively, held the pulse of a steaming cross-section of
      San Franciscans
      `,
    img: 'http://sample.jpg',
    component: 'WorkOne',
    languages: [1, 2],
  },
  updateTargetWork: jest.fn(),
  updateCircle: jest.fn(),
  languages: [
    { id: 1, name: 'Ruby on Rails', icon: 'rails' },
    { id: 2, name: 'React.js', icon: 'react' },
  ],
  history: {
    push: jest.fn(),
  } as any,
  handleClick: jest.fn(),
}

describe('WorkListItem', () => {
  it('should render', () => {
    const wrapper = shallow(<WorkListItem.WorkListItem {...props} />)
    expect(wrapper).toMatchSnapshot()
  })
})
