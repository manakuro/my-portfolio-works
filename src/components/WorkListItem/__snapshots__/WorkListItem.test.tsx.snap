// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkListItem should render 1`] = `
<div
  className="work-list-item"
>
  <div
    className="work-list-item-Card"
  >
    <img
      className="work-list-item-Card_Img"
      onClick={[MockFunction]}
      src="http://sample.jpg"
    />
    <div
      className="work-list-item-Card_Desc"
    >
      <h3
        className="work-list-item-Card_DescHeading"
      >
        EC Website
      </h3>
      <p
        className="work-list-item-Card_DescSubHeading"
      >
        
      Iâ€™ve been a CMT for ten years now, and I have, literally and
      figuratively, held the pulse of a steaming cross-section of
      San Franciscans
      
      </p>
      <div
        className="work-list-item-TechList"
      >
        <LanguageIcon
          key="1"
          language={
            Object {
              "icon": "rails",
              "id": 1,
              "name": "Ruby on Rails",
            }
          }
        />
        <LanguageIcon
          key="2"
          language={
            Object {
              "icon": "react",
              "id": 2,
              "name": "React.js",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;
